from PyQt4 import QtGui, QtCore
from gen.ui_new_project import UiNewProject
from os.path import expanduser
import os


class SettingsView(QtGui.QMainWindow):

    def __init__(self, project_controller):
        super(SettingsView, self).__init__()
        self.project_controller = project_controller
        self.ui = None
        self.build_ui()

    def build_ui(self):
        self.ui = UiNewProject()
        self.ui.setup_ui()
        self.ui.lineEdit_location.setText(expanduser("~"))
        regex = QtCore.QRegExp("\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}")
        validator = QtGui.QRegExpValidator(regex, self.ui.lineEdit_ip)
        self.ui.lineEdit_ip.setValidator(validator)
        self.ui.button_cancel.clicked.connect(self.reject)
        self.ui.button_create.clicked.connect(self.accept)

    def accept(self):
        if self.check_values():
            self.project_controller = self.project_controller.create_project(
                    self.ui.lineEdit_name.text(),
                    self.ui.lineEdit_ip.text(),
                    self.ui.lineEdit_location.text())
            self.ui.done(1)
        else:
            QtGui.QMessageBox.critical(self, "Error", "You need complete the information")
            self.ui.__init__()

    def reject(self):
        self.ui.reject()

    def exec_(self):
        self.ui.exec_()
